### PR TITLE
build_config: add configYamlPath parameter to BuildConfig.parse

### PR DESCRIPTION
**Pull request by kevmoo**
When provided, errors reported when parsing build configuration will
include the build config file path

### CODE REVIEW

File: build_config/lib/src/build_config.dart [hunk 1]
```diff
@@ -43,7 +43,11 @@ class BuildConfig {
     final file = File(configPath);
     if (await file.exists()) {
       return BuildConfig.parse(
-          packageName, dependencies, await file.readAsString());
+        packageName,
+        dependencies,
+        await file.readAsString(),
+        configYamlSourceUrl: file.path,
```
Comment (natebosch) [inline build_config/lib/src/build_config.dart]: Why not `configYamlPath`?

File: build_runner/lib/src/package_graph/build_config_overrides.dart [hunk 2]
```diff
@@ -34,8 +38,12 @@ Future<Map<String, BuildConfig>> findBuildConfigOverrides(
       throw CannotBuildException();
     }
     final yaml = file.readAsStringSync();
-    final config = BuildConfig.parse(packageGraph.root.name,
-        packageGraph.root.dependencies.map((n) => n.name), yaml);
+    final config = BuildConfig.parse(
+      packageGraph.root.name,
+      packageGraph.root.dependencies.map((n) => n.name),
+      yaml,
+      configYamlSourceUrl: file.path,
```
Comment (natebosch) [inline build_runner/lib/src/package_graph/build_config_overrides.dart]: We would need to bump the min version in the pubspec for this.

File: build_config/lib/src/build_config.dart (hunk 1 above)
Comment (kevmoo) [reply build_config/lib/src/build_config.dart]: fixed!

File: build_runner/lib/src/package_graph/build_config_overrides.dart (hunk 2 above)
Comment (kevmoo) [reply build_runner/lib/src/package_graph/build_config_overrides.dart]: yup!

**Comment by kevmoo** [comment]
Before: 

```
[SEVERE] Failed to parse `build.yaml` for json_serializable.
Invalid argument(s): Invalid key(s), could not create BuildConfig.
Supported keys are [builders, post_process_builders, targets, global_options].
line 2, column 1:
  ╷
2 │ _bob: 42
  │ ^^^^
  ╵
```

**Comment by kevmoo** [comment]
After – mucking with builder provided by path dep...

```
[SEVERE] Failed to parse `build.yaml` for json_serializable.
Invalid argument(s): line 2, column 1 of ../json_serializable/build.yaml: Unrecognized keys: [_bob]; supported keys: [builders, post_process_builders, targets, global_options]
  ╷
2 │ _bob: 42
  │ ^^^^
  ╵
```

**Comment by kevmoo** [comment]
Uncle @natebosch 

https://travis-ci.org/dart-lang/build/jobs/527987907#L543

So this is all about some crazy auto-created pubspec something in build_runner tests?

**Comment by natebosch** [comment]
> So this is all about some crazy auto-created pubspec something in build_runner tests?

Yup, we have integration tests which run `pub get` within the test...

There are a few different places with tests like those.

**Comment by kevmoo** [comment]
There's got to be a better way...whatta mess.

Maybe just land this in `build_config` for now and roll build_runner later?

**Comment by natebosch** [comment]
> Maybe just land this in `build_config` for now and roll build_runner later?

:shrug:

**Comment by kevmoo** [comment]
Dropped changes to build_runner...

**Comment by kevmoo** [comment]
@natebosch ...

**Comment by kevmoo** [comment]
@natebosch – all green now CC @jakemac53

**Pull request merged**